system/modules/core/library/Contao/Dbafs.php

```diff
@@ -708,6 +750,33 @@ public static function syncFiles()
 		return $strLog;
 	}
 
+	/**
+	 * Get the folder hash from the databse by combining the hashes of all children
+	 *
+	 * @param string $strPath The relative path
+	 *
+	 * @return string MD5 hash
+	 */
+	public static function getFolderHash($strPath)
+	{
+		$strPath = str_replace(array('\\', '%', '_'), array('\\\\', '\\%', '\\_'), $strPath);
+		$arrHash = [];
+
+		$objChildren = \Database::getInstance()
+			->prepare("SELECT hash, name FROM tl_files WHERE path LIKE ? AND path NOT LIKE ? ORDER BY name")
+			->execute($strPath.'/%', $strPath.'/%/%');
+
+		if ($objChildren !== null)
+		{
+			while ($objChildren->next())
+			{
+				$arrHash[] = $objChildren->hash . $objChildren->name;
```

Comment:
Any particular reason why you have not added a separator here?

Reply:
The hash has always the same length, so there is no need for a seperator.